server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -332,10 +336,13 @@ public RemoteClusterStateManifestInfo writeIncrementalMetadata(
             indicesToBeDeletedFromRemote.remove(indexMetadata.getIndex().getName());
         }
 
-        final DiffableUtils.MapDiff<String, IndexRoutingTable, Map<String, IndexRoutingTable>> routingTableDiff = remoteRoutingTableService
-            .getIndicesRoutingMapDiff(previousClusterState.getRoutingTable(), clusterState.getRoutingTable());
         final List<IndexRoutingTable> indicesRoutingToUpload = new ArrayList<>();
-        routingTableDiff.getUpserts().forEach((k, v) -> indicesRoutingToUpload.add(v));
+        final DiffableUtils.MapDiff<String, IndexRoutingTable, Map<String, IndexRoutingTable>> routingTableIncrementalDiff = remoteRoutingTableService
+            .getIndicesRoutingMapIncrementalDiff(previousClusterState.getRoutingTable(), clusterState.getRoutingTable());
+
+        Map<String, Diff<IndexRoutingTable>> indexRoutingTableDiffs = routingTableIncrementalDiff.getDiffs();
```

Comment:
have we validated this diff object will only contain changed shards and not all shards for the index?

Reply:
Yes, have validated it. Logic to it is defined in diff method for `CUSTOM_ROUTING_TABLE_DIFFABLE_VALUE_SERIALIZER`.